### PR TITLE
Made entire cards clickable

### PR DESCRIPTION
**Pull request by tmott13**
In the OSS Projects Highlights section, made entire cards clickable

Fixes issue: https://github.com/src-d/landing/issues/272

Signed-off-by: Tanya Mott <mott.tanya13@gmail.com>

### CODE REVIEW

**Comment by marnovo** [comment]
Hi @tmott13, thanks for the PR! I've added the appropriate people here to review it.

**Comment by marnovo** [comment]
@ricardobaeta you can pull this branch and test locally.

**Comment by tmott13** [comment]
I need to add the css styling for that, let me do that and I'll update this PR.

**Comment by ricardobaeta** [comment]
Thank you @tmott13 :) Please have in mind https://github.com/src-d/landing/blob/master/static/scss/style.scss

**Comment by tmott13** [comment]
Is it ok to add the rule to the "_cards.scss" file, since it's the card? @ricardobaeta

**Comment by ricardobaeta** [comment]
It makes total sense @tmott13 :) Thank you again.

**Comment by tmott13** [comment]
@ricardobaeta - please review my recent commit "Added style rule to remove underline on links" ...thanks!

**Comment by tmott13** [comment]
@smacker @carlosms - do you guys want to review my changes so this could get merged? Thanks!

**Comment by carlosms** [comment]
Hi,

Thanks for your contribution. This is a good start, but I see some issues.
- Only the highlighted project cards are links, the rest of the page cards are not changed.
- The style changes are not being applied, the `sass` needs to be compiled to `css`.
- Even after compiling the changes in `_cards.scss` the card text style is changed (different `font-weight`).

This is how I would do it instead, which is not so different:

1) Put the `<a>` inside each card, with a new class so we can safely style it later without accidentally overriding the style somewhere else. Like this:
```html
<div class="card text-white orange h-100">
    <a class="card-link" href="https://doc.bblf.sh/">
        <div class="card-body p-5">
            <i class="ic icon-git text-white"></i>
            <p class="card-intro mb-0">Code Analysis</p>
            <h3 class="title-line line-yellow">Babelfish</h3>
            <p>Babelfish is a self-hosted server for source code parsing. It can parse any file, in any
                supported language, extracting an Abstract Syntax Tree (AST) from it and converting it into a
                Universal Abstract Syntax Tree (UAST).</p>
        </div>
    </a>
</div>

...

<div class="card text-white green">
    <a class="card-link" href="https://github.com/src-d/datasets/tree/master/PublicGitArchive">
        <div class="card-body p-4">
            <i class="ic icon-git text-white"></i>
            <p class="card-intro mb-0">Dataset</p>
            <h4 class="m-0">Public Git Archive</h4>
        </div>
    </a>
</div>
```

2) Most of the `card` style is in `static/scss/partials/_main.scss`, so I would add `card-link` there. My goal here is to remove any style for this `<a>`, so I take a look at the main [`a` style here](https://github.com/src-d/landing/blob/master/static/scss/partials/_main.scss#L8), and replace it with `inherit`:
```sass
// Cards

.card {
    border: none;
    border-radius: .25rem;

    .ic {
        font-size: 30px;
        text-decoration: none;
        position: absolute;
        top: 30px;
        right: 30px;
    }

    .author {
        color: $green;
        font-family: 'Cerebri Sans', 'Source Sans Pro', 'Helvetica Neue', Arial, sans-serif;
        font-size: 12px;
        text-transform: uppercase;
        text-decoration: none;
    }

    .card-intro {
        font-size: 13px;
        text-transform: uppercase;

        @media (min-width: 768px) {
            font-size: 18px;
        }
    }

    a.card-link {
        color: inherit;
        font-weight: inherit;
        text-decoration: inherit;

        &:hover {
            color: inherit;
        }
    }
}
```

3) Now that `.scss` file need to be compiled to `css`. Because of some recent changes, we are missing this step in the docs, but it's simple:

- Install sass: https://sass-lang.com/install
- generate css from scss:
```bash
cd static/scss
sass style.scss style.css
```

The PR should also include the changes in `static/scss/style.css` and `static/scss/style.css.map`.

**Comment by tmott13** [comment]
Carlos, thanks for the info! I will make these changes. I was under the impression you guys didn’t want all the cards clickable but I can make them all clickable.

**Comment by tmott13** [comment]
Hi Carlos, 

I made the commits you asked for, sorry for the delay.

Thanks!

File: hugo/layouts/open-source/single.html [hunk 1]
```diff
@@ -150,30 +160,36 @@ <h3 class="title-line text-dark line-yellow mt-5">Models</h3>
             </div>
             <div class="col-md-6 col-lg-4 mb-4">
                 <div class="card text-white green">
-                    <div class="card-body">
-                        <a href="https://github.com/src-d/models#topics" class="ic icon-git text-white"></a>
-                        <p class="card-intro mb-0">Model</p>
-                        <h4 class="m-0">Topic Modeling</h4>
-                    </div>
+                    <a class="card-link" href="https://github.com/src-d/models#topics"></a>
```
Comment (carlosms) [inline hugo/layouts/open-source/single.html]: That extra `</a>` should not be here

File: hugo/layouts/open-source/single.html [hunk 2]
```diff
@@ -214,38 +236,46 @@ <h3 class="title-line text-dark line-yellow mt-5">Code Analysis Tools</h3>
             </div>
             <div class="col-md-6 col-lg-4 mb-4">
                 <div class="card text-white green">
-                    <div class="card-body">
-                        <a href="https://doc.bblf.sh/" class="ic icon-git text-white"></a>
-                        <p class="card-intro mb-0">Code Analysis</p>
-                        <h4 class="m-0">Babelfish</h4>
-                    </div>
+                    <a class="card-link" href="https://doc.bblf.sh/"></a>
```
Comment (carlosms) [inline hugo/layouts/open-source/single.html]: That extra `</a>` should not be here

File: static/scss/partials/_main.scss [hunk 3]
```diff
@@ -115,6 +115,15 @@ a {
             font-size: 18px;
         }
     }
+    a.card-link {
+        color: inherit;
+        font-weight: inherit;
+        text-decoration: inherit;
+
```
Comment (carlosms) [inline static/scss/partials/_main.scss]: After testing it I've realized the `a` element does not fill the whole card if the text is not big enough.
Please add `height: 100%;` here.

File: hugo/layouts/open-source/single.html [hunk 4]
```diff
@@ -76,38 +76,46 @@ <h2 class="text-dark col-md-10 offset-md-1">Open-source components that make mac
                 <h3 class="title-line text-dark line-yellow mt-1 mb-3">OSS Projects Highlights</h3>
             </div>
         </div>
+		
         <div class="row highlight-cards mb-4">
-            <div class="col-lg-4 mb-4 mb-lg-0">
-                <div class="card text-white orange h-100">
+		  <div class="col-lg-4 mb-4 mb-lg-0">
```
Comment (carlosms) [inline hugo/layouts/open-source/single.html]: The new changes use tabs, but the rest of the file is indented using spaced.

File: static/scss/style.css [hunk 5]
```diff
@@ -1,1126 +1,897 @@
...
-.google-calendar iframe {
-  position: absolute;
-  top: 0;
-  left: 0;
-  width: 100%;
-  height: 100%;
-}
-
-/*# sourceMappingURL=style.css.map */
+@charset "UTF-8";
```
Comment (carlosms) [inline static/scss/style.css]: It's strange that this file changed completely, usually it should be a small incremental change.
What is your sass version (`sass --version`)?

File: hugo/layouts/open-source/single.html (hunk 4 above)
Comment (tmott13) [reply hugo/layouts/open-source/single.html]: I don't notice a spacing difference in the file I have? Is this a deal breaker item?

File: static/scss/style.css (hunk 5 above)
Comment (tmott13) [reply static/scss/style.css]: I used Koala to generate the css from the style.scss file. I'm not sure of the version?

**Comment by carlosms** [comment]
Sorry for taking so long to get back to you.
The code looks great, thanks a lot.

Let's get this ready to merge, I can take care of the formatting issues later.
Before we do that I need to ask you the following:
- Please rebase and squash this branch commits into one (a convention described in [our guide](https://github.com/src-d/guide/blob/master/engineering/git-flow.md#3-code-review-and-requested-changes)). If you are not familiar with rebase and squash don't worry, it's simple, see for example [this tutorial](https://www.internalpointers.com/post/squash-commits-into-one-git).
- Make sure all your commits after the rebase contain the [DCO](https://github.com/src-d/guide/blob/e2f42121e158b6cf9e9c307d0ab1decfe45a0245/engineering/documents/CONTRIBUTING.md#certificate-of-origin), some of the current commits do not contain it.

**Comment by tmott13** [comment]
@carlosms  - sorry, I have no idea what I'm doing with this rebase thing. I didn't do my commits thru command line and I'm doing the rebase in the command line. It doesn't show my commits in the log. What do I do?

**Comment by tmott13** [comment]
@carlosms - got it working finally! successfully re-based and squashed my 10 commits into one. Please review and let me know if that works for you. I did add my sign off signature.

**Comment by dpordomingo** [comment]
I'll close this PR since its changes were already included in #309 

Many thanks @tmott13 for helping us to solve #272 addressing all the requests we did during the PR review.

We realized that finally you rebased all changes in https://github.com/tmott13/landing/commit/612af99e459593a9da56a1ee6a35c0f375a7ac6a what was great, but then you merged the old ones again into the current branch https://github.com/tmott13/landing/pull/288/commits/c3d163152d049d0e0cebf6b4e573941a863860a9, so the old ones appeared again. To speed up the things, @carlosms performed the squash and rebase, and addressed his last suggestions via #309 so your work was already merged and will be available in our landing soon.

**Comment by tmott13** [comment]
Thank you all, this was my first PR with source{d} and I learned a lot!

**Comment by carlosms** [comment]
For the sake of completeness, in case you want to dig deeper into git adventures.

As @dpordomingo says commit 612af99 was fine, it had all the changes squashed and was signed-off.
What I did is a [cherry-pick ](https://git-scm.com/docs/git-cherry-pick) of that commit into a new branch of my own fork. This preserves the author of the commit.

Thanks for the contribution.

**Pull request closed without merging**